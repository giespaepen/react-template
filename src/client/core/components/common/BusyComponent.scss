@import "../../../assets/style/AppProperties.scss";
@import "../../../assets/style/AppMixins.scss";

$busy-height: 2px;
$busy-background: white;

.busy {
    background-color:$busy-background;
    height: $busy-height;
    left: 0;
    position: fixed;
    top: 0;
    width: 100%;
    z-index:1000;
    .busy-wrapper {
        width: 100%;
        height:$busy-height;
        .busy-loader{
            display:block;
            height:$busy-height;
            width: 100%;
            background: linear-gradient(
                90deg,
                $color-primary,
                $color-primary 50%,
                $busy-background 50%,
                $busy-background 100%
            );
            background-size:200% 100%;
            -webkit-animation: LoadingAnimation 10s linear infinite;
            -moz-animation: LoadingAnimation 10s linear infinite;
            animation: LoadingAnimation 5s linear infinite;


            span {
                display:none;
            }
        }
    }
}

// Animation definitions
@-webkit-keyframes LoadingAnimation {
    0%{background-position:100% 0%}
    100%{background-position:-100% 0%}
}
@-moz-keyframes LoadingAnimation {
    0%{background-position:100% 0%}
    100%{background-position:-100% 0%}
}
@keyframes LoadingAnimation { 
    0%{background-position:100% 0%}
    100%{background-position:-100% 0%}
}