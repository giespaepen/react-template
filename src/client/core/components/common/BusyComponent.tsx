import "./BusyComponent.scss";
import * as React from "react";
import State from "../../state/BaseState";
import { Action } from "redux";
import { Dispatch } from "redux";
import { IBusyProperties } from "./Properties";
import { connect } from "react-redux";

/**
 * Statefull reactcomponent Busy
 */
@(connect as any)(BusyComponent.mapStateToProps, undefined)
export default class BusyComponent extends React.Component<IBusyProperties, State> {

    public static mapStateToProps(state: State, ownProps: IBusyProperties): IBusyProperties {
        let props: IBusyProperties = Object.assign({}, ownProps);
        props.busy = state.busy;

        return props;
    }

    public render(): any {
        if (this.props.busy) {
            return <div className="busy">
                <div className="busy-wrapper">
                    <div className="busy-loader"><span>Loading</span></div>
                </div>
            </div>;
        } else {
            return <span />;
        }
    }
}