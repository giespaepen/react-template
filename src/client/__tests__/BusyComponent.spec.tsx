import * as React from "react";
import * as renderer from "react-test-renderer";
import BusyComponent from "../core/components/common/BusyComponent";
import { Provider } from "react-redux";
import { applyMiddleware, compose, createStore, Store } from "redux";
import createTestStore from "../core/util/TestStore";
import store from "../core/state/Store";

test("BusyComponent idle snapshot", () => {
    // Arrange
    let state: any = { busy: false };
    let {reducer, store} = createTestStore({ busy: false });

    // Act
    const component = renderer.create(<Provider store={store}><BusyComponent /></Provider>);
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});

test("BusyComponent busy snapshot", () => {
    // Arrange
    const {reducer, store} = createTestStore({ busy: true });
    // Act
    const component = renderer.create(<Provider store={store}><BusyComponent /></Provider>);
    let tree = component.toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
});
