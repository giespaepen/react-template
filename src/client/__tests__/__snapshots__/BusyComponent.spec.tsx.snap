// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusyComponent busy snapshot 1`] = `
<div
  className="busy"
>
  <div
    className="busy-wrapper"
  >
    <div
      className="busy-loader"
    >
      <span>
        Loading
      </span>
    </div>
  </div>
</div>
`;

exports[`BusyComponent idle snapshot 1`] = `<span />`;
